pages/api/id/[form]/submission/confirm.ts

```diff
@@ -0,0 +1,210 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { logMessage } from "@lib/logger";
+import { middleware, cors, jsonValidator, sessionExists } from "@lib/middleware";
+import uuidArraySchema from "@lib/middleware/schemas/uuid-array.schema.json";
+import {
+  DynamoDBDocumentClient,
+  BatchGetCommand,
+  TransactWriteCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { MiddlewareProps, WithRequired } from "@lib/types";
+import { connectToDynamo } from "@lib/integration/dynamodbConnector";
+import { checkOne } from "@lib/cache/flags";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse, props: MiddlewareProps) => {
+  // Is this feature / endpoint active
+  const vaultActive = await checkOne("vault");
+  if (!vaultActive) return res.status(404).json({ error: "Vault not active" });
+
+  const { session } = props as WithRequired<MiddlewareProps, "session">;
+
+  const userEmail = session.user.email;
+  if (userEmail === null)
+    throw new Error(
+      `User does not have an associated email address: ${JSON.stringify(session.user)} `
+    );
+
+  const formId = req.query.form;
+
+  if (Array.isArray(formId) || !formId) return res.status(400).json({ error: "Bad request" });
+
+  const confirmationCodes = req.body as string[];
+
+  try {
+    const dynamoDbClient = connectToDynamo();
+
+    const submissionsFromConfirmationCodes = await getSubmissionsFromConfirmationCodes(
+      formId,
+      confirmationCodes,
+      dynamoDbClient
+    );
+
+    if (submissionsFromConfirmationCodes.submissionsToConfirm.length > 0) {
+      await confirm(formId, submissionsFromConfirmationCodes.submissionsToConfirm, dynamoDbClient);
+    }
+
+    logMessage.info(
+      `user:${userEmail} confirmed form submissions [${submissionsFromConfirmationCodes.submissionsToConfirm.map(
+        (submission) => submission.name
+      )}] for form ID:${formId} at:${Date.now()}`
+    );
+
+    return res.status(200).json({
+      ...(submissionsFromConfirmationCodes.submissionsToConfirm.length > 0 && {
+        confirmedSubmissions: submissionsFromConfirmationCodes.submissionsToConfirm.map(
```

Comment:
I feel like the response JSON attributes I have chosen are not the best so let me know if you have better suggestions

Reply:
IMO they work (naming is hard)

Reply:
I think they work.